spec/models/post_spec.rb

```diff
@@ -0,0 +1,49 @@
+require 'rails_helper'
+
+RSpec.describe Post, type: :model do
+  describe 'For the Post model' do
+    before(:each) do
+      @user = User.new(id: 1, name: 'Rachid', photo: 'image_link', bio: 'full-stack Developer')
+      @post = Post.new(author: @user, title: 'post 1', text: 'post content', comments_counter: 0, likes_counter: 0)
+    end
+
+    it 'if all atributes pass validation' do
+      expect(@post).to be_valid
+    end
+
+    it 'if there is title' do
+      @post.title = nil
+      expect(@post).not_to be_valid
+    end
+
+    it 'if there is max 250 characters' do
+      @post.title = 'Testing lenght'
+      expect(@post).to be_valid
+    end
+
+    it 'if likes counter is integer' do
+      @post.likes_counter = 'string'
+      expect(@post).not_to be_valid
+    end
+
+    it 'if likes counter greater than or equal to zero' do
+      @post.likes_counter = -25
+      expect(@post).not_to be_valid
+    end
+
+    it 'if comments counter greater than or equal to zero.' do
+      @post.comments_counter = -2
+      expect(@post).not_to be_valid
+    end
+
+    it 'if comments counter is integer' do
+      @post.comments_counter = 0.25
+      expect(@post).not_to be_valid
+    end
+
+    it 'if comments counter is integer' do
+      @post.comments_counter = 15
+      expect(@post).to be_valid
+    end
+  end
+end
```

Comment:
- [ ] You should add a test for the `update_posts_counter` and `recent_comments` methods. Kindly add it. [Requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/blog_app_validations_model_spec_n%2B1.md#:~:text=Add%20unit%20specs%20for%20all%20of%20your%20models%27%20methods%20and%20validations.)

Reply:
Addressed